syntax = "proto3";

package tinykv;

service TinyKV {
  rpc Ping (PingRequest) returns (PingResponse) {}
  rpc Put (PutRequest) returns (PutResponse) {}
  rpc Get (GetRequest) returns (GetResponse) {}
}

// MESSAGES

message PingRequest {} 

message PingResponse {
  bool is_ready = 1;
}

message PutRequest {
  string key = 1;
  string val = 2;
  bool is_client = 3;
}

message PutResponse {
  bool operation_success = 1;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string val = 1;
}
